hyperspy/learn/mva.py

```diff
@@ -160,7 +160,7 @@ def decomposition(self,
         """
         to_return = None
         # Check if it is the wrong data type
-        if self.data.dtype.char not in ['e', 'f', 'd']:  # If not float
+        if self.data.dtype.char not in ['e', 'f', 'd', 'F', 'D', 'G']:  # If not float
```

Comment:
Since:
```
import numpy as np
print(np.typecodes["AllFloat"])
>>> 'efdgFDG'
```
Shouldn't this also support `'g'`?

Reply:
@thomasaarholt  pretty sure if you make this change and update it this PR is complete, and only a few years late!